jetty-cdi/src/main/java/org/eclipse/jetty/cdi/CdiServletContainerInitializer.java

```diff
@@ -25,14 +25,24 @@
 
 import org.eclipse.jetty.annotations.AnnotationConfiguration;
 import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.util.Loader;
 import org.eclipse.jetty.util.log.Log;
 import org.eclipse.jetty.util.log.Logger;
 
 /**
- * A {@link ServletContainerInitializer} that introspects for a CDI API
- * implementation within a web application. If the CDI API is found, then
- * a {@link CdiDecorator} is registered as a {@link org.eclipse.jetty.util.Decorator}
- * for the context.
+ * <p>A {@link ServletContainerInitializer} that introspects for a CDI API
+ * implementation within a web application and applies an integration
+ * mode if CDI is found.  CDI integration modes can be selected per webapp with
+ * the "jetty.cdi.mode" init parameter or default to the mode set by the
+ * "jetty.cdi.mode" server attribute.  Supported modes are:</p>
+ * <dl>
+ * <dt>CdiSpiDecorator</dt>
+ *     <dd>Jetty will call the CDI SPI within the webapp to decorate objects (default).</dd>
+ * <dt>CdiDecoratingLister</dt>
+ *     <dd>The webapp may register a decorator on the context attribute
+ *     "org.eclipse.jetty.cdi.decorator".</dd>
+ * </dl>
+ *
  * @see AnnotationConfiguration.ServletContainerInitializerOrdering
  */
 public class CdiServletContainerInitializer implements ServletContainerInitializer
```

Comment:
Now that we removed the checkstyle rule on consecutive upper case letters, should this now be called `CDIServletContainerInitializer` ?

Reply:
Ditto. Happy leave as is.